test/unit/quantum/results/test_frequencies.py

```diff
@@ -9,41 +9,54 @@
 # that they have been altered from the originals.
 
 """Tests for frequency results tools."""
+from test import TYPES
 
-from pytest import mark
-from qiskit.result import Counts
+from numpy import sqrt
+from pytest import mark, raises
+from qiskit.result import Counts, QuasiDistribution
 
 from pr_toolbox.quantum.results.frequencies import (
-    bitflip_counts,
-    bitmask_counts,
-    map_counts,
+    bitflip_frequencies,
+    bitmask_frequencies,
+    convert_counts_to_quasi_dists,
+    map_frequencies,
 )
 
 
 ################################################################################
 ## TESTS
 ################################################################################
-class TestMapCounts:
-    """Test map counts."""
+class TestMapFrequencies:
+    """Test map frequencies."""
 
     @mark.parametrize(
-        "counts, map, expected",
+        "frequencies, map, expected",
         [
-            ({}, lambda _: None, {}),
-            ({0: 1}, lambda _: 0, {0: 1}),
-            ({0: 1}, lambda _: 1, {1: 1}),
-            ({0: 1, 1: 1}, lambda _: 1, {1: 2}),
-            ({0: 0, 1: 1}, lambda k: k + 1, {1: 0, 2: 1}),
+            (Counts({}), lambda _: None, {}),
+            (QuasiDistribution({}), lambda _: None, {}),
+            (Counts({0: 1}), lambda _: 0, {0: 1}),
+            (QuasiDistribution({0: 1}), lambda _: 0, {0: 1}),
+            (Counts({0: 1}), lambda _: 1, {1: 1}),
+            (QuasiDistribution({0: 1}), lambda _: 1, {1: 1}),
+            (Counts({0: 1, 1: 1}), lambda _: 1, {1: 2}),
+            (QuasiDistribution({0: 0.5, 1: 0.5}), lambda _: 1, {1: 1}),
+            (Counts({0: 0, 1: 1}), lambda k: k + 1, {1: 0, 2: 1}),
+            (QuasiDistribution({0: 0, 1: 1}), lambda k: k + 1, {1: 0, 2: 1}),
         ],
     )
-    def test_map_counts(self, counts, map, expected):
-        """Test map counts base functionality."""
-        counts = Counts(counts)
-        assert map_counts(counts, map) == Counts(expected)
+    def test_map_frequencies(self, frequencies, map, expected):
+        """Test map frequencies base functionality."""
+        assert map_frequencies(frequencies, map) == type(frequencies)(expected)
```

Comment:
You also need to check that `shots` and `std_dev` are correctly set for `QuasiDistribution`